docker/scripts/setup_env

```diff
@@ -5,31 +5,89 @@ export LC_ALL=en_US.UTF-8
+# port 3306 and 5432 respectively.
+if [[ -n "${MYSQL_PORT_3306_TCP_ADDR}" || ("${DATABASE_ADAPTER}" == "mysql2" && -n "${DATABASE_HOST}") ]]; then
+    DATABASE_ADAPTER=${DATABASE_ADAPTER:-mysql2}
+    DATABASE_HOST=${DATABASE_HOST:-${MYSQL_PORT_3306_TCP_ADDR}}
+    DATABASE_PORT=${DATABASE_PORT:-${MYSQL_PORT_3306_TCP_PORT:-3306}}
+    DATABASE_ENCODING=${DATABASE_ENCODING:-utf8mb4}
+elif [[ -n "${POSTGRES_PORT_5432_TCP_ADDR}" || ("${DATABASE_ADAPTER}" == "postgresql" && -n "${DATABASE_HOST}") ]]; then
+    DATABASE_ADAPTER=${DATABASE_ADAPTER:-postgresql}
+    DATABASE_HOST=${DATABASE_HOST:-${POSTGRES_PORT_5432_TCP_ADDR}}
+    DATABASE_PORT=${DATABASE_PORT:-${POSTGRES_PORT_5432_TCP_PORT:-5432}}
+    DATABASE_ENCODING=utf8
+else
+    START_MYSQL=${START_MYSQL:-true}
+fi
+
+USE_GRAPHVIZ_DOT=${HUGINN_USE_GRAPHVIZ_DOT:-${USE_GRAPHVIZ_DOT:-dot}}
+DATABASE_HOST=${HUGINN_DATABASE_HOST:-${DATABASE_HOST:-localhost}}
+DATABASE_PORT=${HUGINN_DATABASE_PORT:-${DATABASE_PORT}}
+DATABASE_ENCODING=${HUGINN_DATABASE_ENCODING:-${DATABASE_ENCODING}}
+DATABASE_PASSWORD=${HUGINN_DATABASE_PASSWORD:-${DATABASE_PASSWORD:-password}}
+DATABASE_NAME=${HUGINN_DATABASE_NAME:-${DATABASE_NAME:-huginn_production}}
+RAILS_ENV=${HUGINN_RAILS_ENV:-${RAILS_ENV:-production}}
+
+# Add ENV variables to .env.example which are not present in it but usable
+cat >> /app/.env.example <<EOF
+ASSET_HOST=
+DEFAULT_SCENARIO_FILE=
+RAILS_SERVE_STATIC_FILES=
+SEED_EMAIL=
+SEED_PASSWORD=
+SEED_USERNAME=
+SMTP_OPENSSL_CA_FILE=
+SMTP_OPENSSL_CA_PATH=
+SMTP_OPENSSL_VERIFY_MODE=
+EOF
 
 # Default to the environment variable values set in .env.example
 IFS="="
-grep = /app/.env.example | sed -e 's/^#\([^ ]\)/\1/' | grep -v -e '^#' | \
+sed -n -r -e 's/^#?([A-Za-z0-9_]+ *=.*)/\1/p' /app/.env.example | \
   while read var value ; do
-    eval "echo \"$var=\${$var:-\${HUGINN_$var-\$value}}\""
-  done | grep -v -e ^= > /app/.env
+    eval "echo \"$var=\${HUGINN_$var:-\${$var:-\$value}}\""
+  done > /app/.env
 
-eval "echo PORT=${PORT:-${PORT:-3000}}" >> .env
-eval "echo RAILS_ENV=${RAILS_ENV:-${RAILS_ENV:-production}}" >> .env
-eval "echo RAILS_LOG_TO_STDOUT=true" >> .env
-eval "echo RAILS_SERVE_STATIC_FILES=true" >> .env
+PORT=${HUGINN_PORT:-${PORT:-3000}}
+echo "PORT=$(echo $PORT|awk -F ':' '{print $NF}')" >> .env
+echo "RAILS_ENV=${RAILS_ENV}" >> .env
+echo "RAILS_LOG_TO_STDOUT=true" >> .env
+echo "RAILS_SERVE_STATIC_FILES=true" >> .env
+echo "START_MYSQL=${START_MYSQL}" >> .env
 
 chmod ugo+r /app/.env
+source /app/.env
+
+# use default port number if it is still not set
+case "${DATABASE_ADAPTER}" in
+  mysql2) DATABASE_PORT=${DATABASE_PORT:-3306} ;;
+  postgresql) DATABASE_PORT=${DATABASE_PORT:-5432} ;;
+  *) echo "Unsupported database adapter. Available adapters are mysql2, and postgresql." && exit 1 ;;
+esac
+
+bundle install --without test development --path vendor/bundle
+
+# Configure the unicorn server
+mv config/unicorn.rb.example config/unicorn.rb
```

Comment:
This line fails with `mv: cannot stat ‘config/unicorn.rb.example’: No such file or directory` when a container is restarted. Simply copying instead of moving should fix it.

Reply:
Ah, you're right, I forgot to change this when I moved it from `setup`.